scales-xml/src/main/scala/org/http4s/scalesxml/ScalesInstances.scala

```diff
@@ -0,0 +1,38 @@
+package org.http4s
+package scalesxml
+
+import java.io.ByteArrayInputStream
+
+import org.http4s.headers.`Content-Type`
+import org.xml.sax.{InputSource, SAXParseException}
+
+import scala.util.control.NonFatal
+import scalaz.concurrent.Task
+import scales.xml.serializers.{SerializeableXml, SerializerFactory}
+import scales.xml.{asString, Doc, loadXml, XmlVersion}
+
+
+trait ScalesInstances {
+
+  implicit def xmlDocEncoder(implicit charset: Charset = DefaultCharset, serf : SerializerFactory, sxml : SerializeableXml[Doc]): EntityEncoder[Doc] =
+    EntityEncoder.stringEncoder(charset)
+      .contramap[Doc](xml => asString(xml))
+      .withContentType(`Content-Type`(MediaType.`application/xml`))
+
+
+  implicit def xmlDocDecoder(implicit xmlVer: XmlVersion): EntityDecoder[Doc] = {
+    import EntityDecoder._
+    decodeBy(MediaType.`text/xml`, MediaType.`text/html`, MediaType.`application/xml`){ msg =>
+      collectBinary(msg).flatMap[Doc] { arr =>
```

Comment:
Instead of materializing the whole body and parsing it at once, what if this used the aalto module for incremental pull parsing?  That's something I've wanted to add to the scala-xml module for a while (#153).

That code would probably not live here, but be added to Scales as its scalaz-stream integration.  This could then remain a very light glue module.